extensions/liquibase-mongodb/deployment/src/main/java/io/quarkus/liquibase/mongodb/deployment/LiquibaseMongodbProcessor.java

```diff
@@ -154,82 +168,43 @@ void nativeImageConfiguration(
-                liquibase.serializer.SnapshotSerializer.class,
-                liquibase.servicelocator.ServiceLocator.class,
-                liquibase.snapshot.SnapshotGenerator.class,
-                liquibase.sqlgenerator.SqlGenerator.class,
-                liquibase.structure.DatabaseObject.class,
-                liquibase.logging.mdc.MdcManager.class)
-                .forEach(t -> addService(services, reflective, t, false));
-
         // Register Precondition services, and the implementation class for reflection while also registering fields for reflection
-        addService(services, reflective, liquibase.precondition.Precondition.class, true);
+        addService(services, reflective, liquibase.precondition.Precondition.class.getName(), true);
 
         // CommandStep implementations are needed (just like in non-mongodb variant)
-        addService(services, reflective, liquibase.command.CommandStep.class, false,
+        addService(services, reflective, liquibase.command.CommandStep.class.getName(), false,
                 "liquibase.command.core.StartH2CommandStep");
 
-        // liquibase XSD
-        resource.produce(new NativeImageResourceBuildItem(
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.7.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.8.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.9.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.10.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.11.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.12.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.13.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.14.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.15.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.16.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.17.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.18.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.19.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.20.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.21.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.22.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.23.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.24.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.25.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd",
-                "liquibase.build.properties"));
+        var dependencies = curateOutcome.getApplicationModel().getDependencies();
```

Comment:
AFAICS, there are two places where the new `ofDependencyResources` method is used.

Just to clarify, are all dependencies instead of only the runtime ones scanned on purpose in both cases?

Reply:
You're right, we could reduce the scope to the runtime dependencies. I'll take care of that.